Sweep_line_2/include/CGAL/Sweep_line_2/Sweep_line_curve_pair.h

```diff
@@ -220,4 +224,20 @@ class random_access_input_iterator
 
 } //namespace CGAL
 
+
+namespace boost {
+  
+  template <class Subcurve_>
+  struct hash<CGAL::Curve_pair<Subcurve_> >
+    :  public CGAL::unary_function<CGAL::Curve_pair<Subcurve_>, std::size_t>  {
+
+    std::size_t operator()(const CGAL::Curve_pair<Subcurve_>& cp) const
+    {
+      return boost::hash<std::pair<Subcurve_*, Subcurve_*> >()(cp.pair());
+    }
```

Comment:
The documentation of `boost::hash` does not say it is allowed to specialize it. What is documented is [the creation of an overload of  `hash_value`][*], found by ADL.

[*]: http://www.boost.org/doc/libs/1_65_1/doc/html/hash/custom.html